data/scripts/check-samples-dependencies.py

```diff
@@ -0,0 +1,165 @@
+    return samplesData
+
+def getNodeData(path):
+    with open(path, 'r') as f:
+        #Skip comments
+        fbpContent = re.sub('^#.*', '', f.read(), 0, re.M).strip()
+    #Skip declares
+    lastDeclare = fbpContent.rfind('DECLARE')
+    if lastDeclare > -1:
+        nodes = nodeTypeRegex.findall(fbpContent[fbpContent.index('\n', lastDeclare):])
+    else:
+        nodes = nodeTypeRegex.findall(fbpContent)
+    declaredTypes = declaredTypesRegex.findall(fbpContent)
+    toRemove = []
+    extraFbps = []
+    extraDeps = []
+    for declaredType in declaredTypes:
+        t = declaredType[1].strip()
+        if  t == 'composed-new' or t == 'composed-split':
+            toRemove.append(declaredType[0].strip())
+        elif t == 'js':
+            toRemove.append(declaredType[0].strip())
+            if 'flow_metatype_javascript' not in extraDeps:
+                extraDeps.append('flow_metatype_javascript')
+        elif t == 'http-composed-server':
+            toRemove.append(declaredType[0].strip())
+            if 'flow_metatype_http_composed_server' not in extraDeps:
+                extraDeps.append('flow_metatype_http_composed_server')
+        elif t == 'http-composed-client':
+            toRemove.append(declaredType[0].strip())
+            if 'flow_metatype_http_composed_client' not in extraDeps:
+                extraDeps.append('flow_metatype_http_composed_client')
+        elif t == 'fbp':
+            toRemove.append(declaredType[0].strip())
+            extraFbps.append(declaredType[2].strip())
+    return (list(filter(lambda x: False if x in toRemove else True, nodes)), extraFbps, extraDeps)
+
+def hasDep(deps, node):
+    for dep in deps:
+        if node in dep:
+            return True
+    return False
+
+def checkFbpDeps(rootPath, samplesData, fbpFiles):
+    for fbp in fbpFiles:
+        if fbp not in samplesData:
+            continue
+        fbpPath = os.path.join(rootPath, fbp)
+        nodeData = getNodeData(fbpPath)
+        nodes = list(map(lambda x: samplesData[fbp]['node_aliases'][x].lower() if x in samplesData[fbp]['node_aliases'] else x.lower(),nodeData[0]))
+        #Extra deps.
+        if len(nodeData[2]) > 0:
+            nodes.extend(nodeData[2])
+        #Declared FBPs
+        if len(nodeData[1]) > 0:
+            for child in nodeData[1]:
+                samplesData[child] = samplesData[fbp]
+            checkFbpDeps(rootPath, samplesData, nodeData[1])
+        missing = list(filter(lambda x: hasDep(samplesData[fbp]['deps'], x.replace('-', '_')) == False, nodes))
+        if len(missing) == 0:
+            continue
+        problems['missing_deps'][fbpPath] = missing
+
+def loadBlackList(path):
+    with open(path, 'r') as f:
+        return json.load(f)
+
+if __name__ == "__main__":
+    exitWithError = False
+
+    if len(sys.argv) < 3:
+        print('Usage: ./check-sample-dependencies samples_root_dir black_list_dir')
```

Comment:
mmm... here you say black_list_dir, but actually you're waiting for apath for a json file, right?